qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -292,67 +324,153 @@ def circuits(self) -> List[QuantumCircuit]:
+                # Assign duration in dt to create pulse schedule.
+                assigned_circ = circ.assign_parameters(
+                    {param_duration: valid_duration_dt},
+                    inplace=False,
+                )
+                assigned_circ.metadata["xval"] = self.num_pulses * flat_top_width_sec
+                circs.append(assigned_circ)
+
+        return circs
+
+    def _unitary_circuits(self):
+        """Protocol to create circuits with unitary gate.
+
+        Unitary gate has no timing constraints and accepts duration in sec.
+        Basically, this method doesn't require backend apart from conversion of
+        sigma in samples into sec.
+        """
+        # Assume this parameter is in units of sec.
+        param_duration = circuit.Parameter("duration")
+
+        # Gate duration will be shown in sec, which is more intuitive.
+        cr_gate = self._gate_cls(width=self._get_width(param_duration))
+
+        # Create parameterized circuits without calibration.
+        tmp_circs = []
+        for control_state in (0, 1):
+            for meas_basis in ("x", "y", "z"):
+                tmp_qc = QuantumCircuit(2, 1)
+                if control_state:
+                    tmp_qc.x(0)
+                tmp_qc.compose(
+                    other=self._build_cr_circuit(cr_gate),
+                    qubits=[0, 1],
+                    inplace=True,
+                )
+                if meas_basis == "y":
+                    tmp_qc.sdg(1)
+                if meas_basis in ("x", "y"):
+                    tmp_qc.h(1)
+                tmp_qc.measure(1, 0)
+                tmp_qc.metadata = {
+                    "control_state": control_state,
+                    "meas_basis": meas_basis,
+                }
+                tmp_circs.append(tmp_qc)
+
+        circs = []
+        for duration in self._get_durations():
+            flat_top_width_sec = self._get_width(duration)
+            if flat_top_width_sec < 0:
+                raise ValueError(
+                    f"Input duration={duration} is less than pulse ramps lengths, resulting in "
+                    f"a negative flat top length of {flat_top_width_sec} sec. "
+                    f"This cross resonance schedule is invalid."
+                )
+
+            for circ in tmp_circs:
+                # Assign duration in sec since this is unitary gate.
+                assigned_circ = circ.assign_parameters(
+                    {param_duration: duration},
+                    inplace=False,
+                )
+                assigned_circ.metadata["xval"] = self.num_pulses * flat_top_width_sec
+                circs.append(assigned_circ)
+
+        return circs
 
     def _finalize(self):
         """Set analysis option for initial guess that depends on experiment option values."""
         edge_duration = np.sqrt(2 * np.pi) * self.experiment_options.sigma * self.num_pulses
+        timing = BackendTiming(self.backend)
```

Comment:
You initialize this multiple times in this class. Why not make it a class variable, e.g. `self._backend_timing`? I think this would also mean wrapping the backend setter of the base class to make sure that `self._backend_timing` is always consistent with `self._backend`.

Reply:
Good point, done in https://github.com/Qiskit/qiskit-experiments/pull/933/commits/1f1bd5a838aed7a50d1f6a42fe5b71d592c52396